src/3d/qgs3dutils.cpp

```diff
@@ -1026,3 +1026,38 @@ QgsPoint Qgs3DUtils::screenPointToMapCoordinates( const QPoint &screenPoint, con
   const QgsPoint mapPoint( mapTransform.x(), mapTransform.y(), mapTransform.z() );
   return mapPoint;
 }
+
+QList<QVector4D> Qgs3DUtils::lineSegmentToClippingPlanes( const QgsVector3D &point1, const QgsVector3D &point2, const double distance, const QgsVector3D &origin )
```

Comment:
According to the docstring, the `distance` is supposed to be positive. This should be checked at the beginning of the function.